x-pack/plugins/observability_solution/metrics_data_access/common/inventory_models/types.ts

```diff
@@ -235,29 +235,34 @@ export type TSVBMetricModelCreator = (
   interval: string
 ) => TSVBMetricModel;
 
-export const ESBasicMetricAggRT = rt.record(
-  rt.string,
-  rt.union([
-    rt.undefined,
-    rt.type({
-      field: rt.string,
-    }),
-  ])
-);
-
-export const ESDerivativeAggRT = rt.type({
-  derivative: rt.type({
-    buckets_path: rt.string,
-    gap_policy: rt.keyof({ skip: null, insert_zeros: null }),
-    unit: rt.string,
-  }),
-});
+export const isBasicMetricAgg = (
+  agg: unknown
+): agg is Record<string, undefined | Pick<estypes.AggregationsMetricAggregationBase, 'field'>> => {
+  if (agg && typeof agg === 'object' && Object.keys(agg).length !== 0) {
```

Comment:
Q: Do we need the `typeof agg === 'object'` check here when we already check if agg is defined? I mean is there an option not to be an object?

Reply:
I wanted to make it future-proof. Technically, we expect the unexpected (`unknown`) so this check might be run against anything. With `typeof agg === 'object'` I intended to avoid further proceeding if `agg` is not `object`. Wdyt?

Reply:
Sounds good, let's keep the object check 👍